.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - prod-*
+      - stable-*
+
+  pull_request:
+    branches:
+      - main
+      - master
+      - prod-beta
+      - prod-stable
+      - stage-beta
+      - stage-stable
+env:
+  BRANCH: ${{ github.base_ref }}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
```

Comment:
Why do we need separate jobs for lint and units? It runs thru the many of the same steps twice

Reply:
Separate jobs lets them run concurrently, reducing overall runtime. There are some duplicate steps, but that's because the setup needs to happen on each runner.

Reply:
We could avoid the setup steps by using our own container image for CI runs. That's a much more reliable (and faster) way to run tests, but requires maintaining the CI image(s).